packages/dialog/src/DialogWrapper.ts

```diff
@@ -163,161 +97,76 @@ export class DialogWrapper extends FocusVisiblePolyfillMixin(SpectrumElement) {
         );
     }
 
-    protected handleClose(event: Event): void {
-        event.stopPropagation();
-        this.close();
-    }
-
-    public close(): void {
-        this.open = false;
-    }
-
-    private dispatchClosed(): void {
-        this.dispatchEvent(
-            new Event('close', {
-                bubbles: true,
-            })
-        );
-    }
-
-    protected handleUnderlayTransitionend(event: TransitionEvent): void {
-        if (!this.open && event.propertyName === 'visibility') {
-            this.dispatchClosed();
-            this.resolveTransitionPromise();
-        }
-    }
-
-    protected handleModalTransitionend(): void {
-        if (this.open || !this.underlay) {
-            this.resolveTransitionPromise();
-            if (!this.open) {
-                this.dispatchClosed();
-            }
-        }
-    }
-
-    protected override update(changes: PropertyValues<this>): void {
-        if (changes.has('open') && changes.get('open') !== undefined) {
-            this.transitionPromise = new Promise(
-                (res) => (this.resolveTransitionPromise = res)
-            );
-        }
-        super.update(changes);
-    }
-
-    protected override render(): TemplateResult {
+    protected override renderDialog(): TemplateResult {
         return html`
-            ${this.underlay
-                ? html`
-                      <sp-underlay
-                          ?open=${this.open}
-                          @click=${this.dismiss}
-                          @transitionend=${this.handleUnderlayTransitionend}
-                      ></sp-underlay>
-                  `
-                : html``}
-            <div
-                class="modal ${this.mode}"
-                @transitionend=${this.handleModalTransitionend}
+            <sp-dialog
+                ?dismissable=${this.dismissable}
+                ?no-divider=${this.noDivider}
+                ?error=${this.error}
+                mode=${ifDefined(this.mode ? this.mode : undefined)}
```

Comment:
Can this not just be `ifDefined(this.mode)`?

Reply:
Cruft from way back to the initial creation of the element 🤯 updated.